### ISSUE TITLE
Virtuals in nested objects inside arrays don't work

### ISSUE THREAD

**Issue by cur3n4** [neumino/thinky#375]
```javascript
var Person = thinky.createModel("Person", {
	address : [
		{
			street: String,
			fullName: type.virtual().default(function() {
				return "My full address";
			})
		}
	]
});

var person = new Person({
        address : [ { street: 'Lodge' } ]
});

```

When saving person this will cause a [TypeError: Cannot read property '0' of undefined]

**Comment by neumino** [comment]
Thanks @cur3n4 for the snippet to reproduce this bug.
It's fixed in 2.1.11

**Comment by cur3n4** [comment]
Great! Thanks a lot
